src/StepDefinition/MinkElementTrait.php

```diff
@@ -185,4 +185,59 @@ public function textAppearsTimesInElement($text, $element, $number, $selector =
     }
   }
 
+  /**
+   * @Then /^(?:|I )wait for (?P<num>\d+) "(?P<element>[^"]*)" elements to appear?$/
+   *
+   * @param string $element
+   * @param int $num
+   *
+   * @throws \Exception
+   */
+  public function iWaitForElementToAppear($num, $element) {
+    $this->spinFunction(function ($context) use ($element, $num) {
+      try {
+        $this->assertSession()->elementsCount('css', $element, intval($num));
+        return TRUE;
+      }
+      catch (ExpectationException $e) {
+        // NOOP
+      }
+      return FALSE;
+    });
+  }
+
+  /**
+   * Implementation of spin method.
+   *
+   * @param string $lambda
+   *   Function expresion.
+   *
+   * @return bool
+   *
+   * @throws \Exception
+   */
+  public function spinFunction($lambda) {
```

Comment:
Can we piggy-back upon the existing waitFor() funciton here?
Anyway, I think we should have the maximum wait time as argument/configureable and set  this to 5 seconds only. Then I'd re-check at least every ~300ms since it should be done after ~300ms when things are fast.